ios/viewmodels/view-model-example.md

```diff
@@ -0,0 +1,280 @@
+        didSet {
+            viewModels = [SavedPreztoViewModel?](count: savedPreztos.count, repeatedValue: nil)
+        }
+    }
+
+    var viewModels = [SavedPreztoViewModel?]()
+
+    func viewModelForPreztoAtIndex(index: Int) -> SavedPreztoViewModel {
+        if let viewModel = viewModels[index] {
+            return viewModel
+        } else {
+            let viewModel = SavedPreztoViewModel(prezto: savedPreztos[index])
+            viewModels[index] = viewModel
+            return viewModel
+        }
+    }
+}
+```
+
+The `SavedPreztoViewModel` encapsulates view styling logic for individual cells, as well as the logic to download thumbnail images.
+
+### SavedPreztoViewModel.swift
+
+```swift
+class SavedPreztoViewModel {
+
+    let prezto: Prezto
+    var images: [UIImage]?
+
+    init(prezto: Prezto) {
+        self.prezto = prezto
+    }
+
+    var titleLabelText: String {
+        return prezto.title ?? ""
+    }
+
+    var accessCodeLabelText: String {
+        guard let code = prezto.accessCode else {
+            return ""
+        }
+        return "Access code: \(code)"
+    }
+
+    var modifiedLabelText: String {
+        guard let updatedAt = prezto.updatedAt else {
+            return ""
+        }
+        return "Modified on \(updatedAt.dateInLocalTimeZone())"
+    }
+
+    func getImages(completion: () -> Void) {
+        if let _ = images {
+            completion()
+        } else {
+            prezto.thumbnailImages { images in
+                self.images = images
+                completion()
+            }
+        }
+    }
+```
+
+## Using the view models
+
+Below is a stripped-down version of `SavedPreztosViewController` showing the relevant places where the view models are used:
+
+### SavedPreztosViewController.swift
+
+```swift
+class SavedPreztosViewController: UIViewController, UITableViewDataSource, UITableViewDelegate, UIScrollViewDelegate, SWTableViewCellDelegate {
+    
+    @IBOutlet weak var tableView: UITableView!
+
+    let viewModel = SavedPreztosViewModel()
+
+    /********************************************************************
+     * Skipping a bunch of view controller setup and life-cycle code... *
+     ********************************************************************/
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+
+        viewModel.clearCachedImages()
+    }
+    
+    /****************************************
+     * Skipping a bunch of UI setup code... *
+     ****************************************/
+
+    private func refreshSavedPreztosList() {
+        viewModel.retrievePreztos { result in
+            switch result {
+            case .Success:
+                dispatch_async(dispatch_get_main_queue()) {
+                    self.tableView.reloadData()
+                    self.setupBunnyImage()
+                }
+            case .Failure(let error):
+                // TODO: Need UI here to retry
```

Comment:
For the purposes of this example, rather than have a `TODO` let's just call a method that doesn't actually exists?  E.g., `presentRefreshFailedAlert()`